src/frontend/src/lib/types/errors.ts

```diff
@@ -1 +1,13 @@
+import type { TokenId } from '$lib/types/token';
+
 export class UserProfileNotFoundError extends Error {}
+
+export class AddressError extends Error {
+	tokenId: TokenId;
+
+	constructor(tokenId: TokenId) {
+		super();
+		Object.setPrototypeOf(this, AddressError.prototype);
```

Comment:
I honestly don't know what's the goal of `Object.setPrototypeOf...`. What's the oucome if I may ask for my knowledges?

Reply:
usually it is suggested to use it right after a `super()` statement: I have found a sort of technical explanation [here](https://stackoverflow.com/a/76851585), but reading better it says that it is indeed not needed for instances of `Error`

So I am removing it

Reply:
Thanks for the explanation, did not knew. Somehow always expected the bundler to take care of those kind of `setPropotyOf` thingy. Good to know!